src/test/javascript/spec/component/programming-exercise/feedback-analysis/modals/feedback-filter-modal.component.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { FeedbackFilterModalComponent } from 'app/exercises/programming/manage/grading/feedback-analysis/Modal/feedback-filter-modal.component';
+import { LocalStorageService } from 'ngx-webstorage';
+import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { TranslateModule } from '@ngx-translate/core';
+import { ArtemisSharedCommonModule } from 'app/shared/shared-common.module';
+import { RangeSliderComponent } from 'app/shared/range-slider/range-slider.component';
+
+describe('FeedbackFilterModalComponent', () => {
+    let fixture: ComponentFixture<FeedbackFilterModalComponent>;
+    let component: FeedbackFilterModalComponent;
+    let localStorageService: LocalStorageService;
+    let activeModal: NgbActiveModal;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [TranslateModule.forRoot(), ReactiveFormsModule, ArtemisSharedCommonModule, RangeSliderComponent, FeedbackFilterModalComponent],
+            providers: [{ provide: LocalStorageService, useValue: { store: jest.fn(), clear: jest.fn(), retrieve: jest.fn() } }, NgbActiveModal, FormBuilder],
+        }).compileComponents();
+
+        fixture = TestBed.createComponent(FeedbackFilterModalComponent);
+        component = fixture.componentInstance;
+        localStorageService = TestBed.inject(LocalStorageService);
+        activeModal = TestBed.inject(NgbActiveModal);
+        component.minCount.set(0);
+        component.maxCount.set(10);
+        fixture.detectChanges();
+    });
+
+    it('should initialize filter form correctly', () => {
+        expect(component.filterForm).toBeDefined();
+        component.filterForm.get('occurrence')?.setValue([component.minCount(), component.maxCount()]);
+        expect(component.filterForm.value).toEqual({
+            tasks: [],
+            testCases: [],
+            occurrence: [0, 10],
+        });
+    });
+
+    it('should call localStorage store when applying filters', () => {
+        const storeSpy = jest.spyOn(localStorageService, 'store');
+        const emitSpy = jest.spyOn(component.filterApplied, 'emit');
+        const closeSpy = jest.spyOn(activeModal, 'close');
+        component.maxCount.set(0);
+        component.maxCount.set(10);
+        component.filterForm.get('occurrence')?.setValue([component.minCount(), component.maxCount()]);
+        component.applyFilter();
+
+        expect(storeSpy).toHaveBeenCalledWith(component.FILTER_TASKS_KEY, []);
```

Comment:
can you be more specific through out the file

```suggestion
        expect(storeSpy).toHaveBeenCalledExactlyOnceWith(component.FILTER_TASKS_KEY, []);
```

Reply:
its called 3 times in the lines below aswell, thats why this wont work